include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -251,6 +251,36 @@ def ConcatOp : VariadicRTLOp<"concat"> {
   ];
 }
 
+def ArraySliceOp : RTLOp<"slice", [NoSideEffect]> {
+  let summary = "Get a range of values from an array";
+  let description = [{
+    Extracts a sub-range from an array. The range is from `lowIndex` to
+    `lowIndex` + the number of elements in the return type, non-inclusive on
+    the high end, consistent with common software function calls (e.g.
+    `memcpy`). For instance,
```

Comment:
Please don't mention memcpy here, this raises the spectre of a bunch of alignment and padding issues that don't exist here.  Just say it extracts out N elements starting from M.

Reply:
I'd also mention that the index width is fixed with a specific algorithm.  Also mention that out of range indexes are undefined behavior.